public-ui/src/app/views/survey-view/survey-view.component.ts

```diff
@@ -278,17 +277,21 @@ export class SurveyViewComponent implements OnInit, OnDestroy {
   }
 
   public filterResults() {
+    if (this.searchText.value) {
+      window['dataLayer'].push({
```

Comment:
seems like it would be cleaner to create a central utility which initialized the dataLayer object and provided a helper for pushing a gtag event

Reply:
Moved this to db.config which is for now central and being used by every component.